src/workerd/api/crypto-impl-asymmetric.c++

```diff
@@ -1024,9 +1024,16 @@ public:
     auto& publicKeyImpl = kj::downcast<EllipticKey>(*publicKey->impl);
 
     // Adapted from https://wiki.openssl.org/index.php/Elliptic_Curve_Diffie_Hellman:
-    auto privateEcKey = EVP_PKEY_get0_EC_KEY(getEvpPkey());
-    auto publicEcPoint = EC_KEY_get0_public_key(EVP_PKEY_get0_EC_KEY(publicKeyImpl.getEvpPkey()));
-    auto fieldSize = EC_GROUP_get_degree(EC_KEY_get0_group(privateEcKey));
+    auto& privateEcKey = JSG_REQUIRE_NONNULL(EVP_PKEY_get0_EC_KEY(getEvpPkey()),
```

Comment:
Hmm, KJ's REQUIRE_NONNULL macro (which JSG_REQUIRE_NONNULL is based on) was intended to be used with `kj::Maybe`, but here you are using them with raw pointers. This isn't actually intended to work... I am not really sure whether to say we shouldn't be using it this way, or just go ahead and sanction this as an allowed use... I guess it's reasonably intuitive?

Reply:
heh, I thought the same and had to go check to make sure it would actually work. I'm fine with this since it does seem fairly intuitive but would leave that ultimately up to you from a kj perspective.

Reply:
It is pretty intuitive, in this case it allows accurately reporting the error without too much code. I guess an alternative would be to have a similar libkj macro that omits the KJ_REQUIRE_NONNULL's `readMaybe` step

The change is largely based on lines 1207ff, where the same approach is used, so it seems to work https://github.com/cloudflare/workerd/blob/bdfb3cde62d5a91b5e5ef0b34b554ed8c3ef4568/src/workerd/api/crypto-impl-asymmetric.c%2B%2B#L1207